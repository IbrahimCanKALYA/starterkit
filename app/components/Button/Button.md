# Button

## Description

- Simple button component without any spesific properties.

## Funtions

- absent !

## Props

- text(String) :  Text to be displayed center of the button.

- buttonTextStyle(Object) : Style object to be overrided button text styles.

- buttonStyle(Object) : Style object to be overrided button container styles.