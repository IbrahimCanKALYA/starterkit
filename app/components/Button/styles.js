/* @flow */
const styles = {
  container: {
    flexDirection: 'row',
    minHeight: 50,
    borderRadius: 3,
    backgroundColor: '#007aff',
  },
  buttonText: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
};

export default styles;
