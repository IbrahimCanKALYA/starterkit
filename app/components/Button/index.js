/* @flow */
import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

type Props = any;

class Button extends Component<Props> {
  static defaultProps: any;

  render() {
    const {
      props: { text, buttonStyle, buttonTextStyle },
      props,
    } = this;
    return (
      <TouchableOpacity style={[styles.container, buttonStyle]} {...props}>
        <View style={styles.buttonText}>
          <Text style={buttonTextStyle}>{text}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string,
  buttonTextStyle: PropTypes.object,
  buttonStyle: PropTypes.object,
};
Button.defaultProps = {
  text: 'Button',
  buttonTextStyle: {},
  buttonStyle: {},
};

export default Button;
