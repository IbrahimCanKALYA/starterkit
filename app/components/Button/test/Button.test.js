import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

test('Button renders correctly', () => {
  const tree = renderer.create(<Button text="Button" buttonStyle={{}} buttonTextStyle={{}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
