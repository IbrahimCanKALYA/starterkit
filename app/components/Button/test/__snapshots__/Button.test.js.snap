// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#007aff",
      "borderRadius": 3,
      "flexDirection": "row",
      "minHeight": 50,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={Object {}}
    >
      Button
    </Text>
  </View>
</View>
`;
